lib/config/config.js

```diff
@@ -439,6 +455,101 @@ function loadConfigFile (configFile) {
   return null
 }
 
+/**
+ * Collects information about the agent, environment and configuration options that are
+ * sets from a source (env vars, start options or config file). For each config option it
+ * collects
+ * - source: where te value came from (environment, start, file)
+ * - sourceValue: the raw value provided im the source
+ * - normalizedValue: the value normalized by the configuration which is used in the agent
+ * - file: if source === 'file' this proprty is defined with the path of the config file
+ *
+ * @param {Object} config The configuration object with normalized options
+ * @param {Array<{ name: string, ops: Object}>} sources The list of sources of options ordered by priority
+ * @param {String | undefined} configFilePath Path of the config file if loaded
+ * @returns {Object} Object with agent, environment, and configuration data.
+ */
+function getLoggingPreambleData (config, sources, configFilePath) {
+  const optsFromSources = sources.reduce((prev, s) => Object.assign({}, s.opts, prev), {})
+  const result = {
+    agentVersion: AGENT_VERSION,
+    env: {
+      pid: process.pid,
+      pname: process.title,
+      os: os.release(),
+      arch: os.arch(),
+      host: os.hostname(),
+      timezone: getTimezoneUtc(),
+      runtime: `Node.js ${process.version}`
+    },
+    config: {}
+  }
+
+  const mandatoryKeys = ['serviceName', 'serviceVersion', 'serverUrl', 'logLevel']
+  const configKeys = new Set(mandatoryKeys.concat(Object.keys(optsFromSources)))
+
+  configKeys.forEach(key => {
+    if (key in EXCLUDE_FIELDS) {
+      return
+    }
+
+    const shouldRedact = key in REDACT_FIELDS
+    const sourceObj = sources.find(s => (key in s.opts))
+    const sourceName = (sourceObj && sourceObj.name) || 'default'
+
+    result.config[key] = {
+      normalizedName: formatOptionKey(key),
+      source: sourceName,
+      sourceValue: formatOptionValue(optsFromSources[key], shouldRedact),
+      normalizedValue: formatOptionValue(config[key], shouldRedact)
+    }
+    if (sourceName === 'file') {
+      result.config[key].file = configFilePath
+    }
+  })
+
+  return result
+}
+
+/**
+ * Formats the config value to be printed in log. For now it quotes the string values
+ *
+ * @param {any} value The value to be formatted if it's a string
+ * @param {Boolean} shouldRedact true if the value should be redacted
+ * @returns {any}
+ */
+function formatOptionValue (value, shouldRedact) {
+  if (shouldRedact) {
+    return REDACTED
+  }
+
+  return typeof value === 'string' ? `"${value}"` : value
```

Comment:
The spec is enforcing it but you're right. I guess is because the spec is assuming a string format (but not enforcing it) instead of a JSON one.
https://github.com/elastic/apm/blob/9fadb845e0f7d7c4da544719c734f3ebedf8fd8a/specs/agents/logging.md?plain=1#L138